verify-service/src/main/java/io/mosip/verifyservice/services/VerifiablePresentationSubmissionServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+            Algorithm algorithm = Algorithm.RSA256(getPublicKeyFromString(publicKeyPem), null);
+            JWTVerifier verifier = JWT.require(algorithm).build();
+            verifier.verify(jws);
+
+            JSONArray verifiableCredentials = new JSONObject(vpSubmissionDto.getVpToken()).getJSONArray(Constants.KEY_VERIFIABLE_CREDENTIAL);
+            List<VcResult> verificationResults = getVcVerificationResults(verifiableCredentials);
+            boolean combinedVerificationStatus = true;
+            for (VcResult verificationResult : verificationResults) {
+                combinedVerificationStatus = combinedVerificationStatus && (verificationResult.getVerificationStatus() == VerificationStatus.SUCCESS);
+            }
+            if (!combinedVerificationStatus) {
+                throw new VerificationFailedException();
+            }
+            vpSubmissionRepository.save(new VpSubmission(vpSubmissionDto.getState(), vpSubmissionDto.getVpToken(), vpSubmissionDto.getPresentationSubmission(), SubmissionStatus.SUCCESS));
+        } catch (Exception e) {
+            e.printStackTrace();
+            vpSubmissionRepository.save(new VpSubmission(vpSubmissionDto.getState(), vpSubmissionDto.getVpToken(), vpSubmissionDto.getPresentationSubmission(), SubmissionStatus.FAILED));
+        }
+
+        authorizationRequestCreateResponseRepository.findById(vpSubmissionDto.getState()).map(authorizationRequestCreateResponse -> {
+            authorizationRequestCreateResponse.setSubmissionState(SubmissionState.COMPLETED);
+            authorizationRequestCreateResponseRepository.save(authorizationRequestCreateResponse);
+            return null;
+        });
+    }
+
+    @Override
+    public VpTokenResultDto getSubmissionResult(String requestId, String transactionId) {
+        VpSubmission vpSubmissionResult = vpSubmissionRepository.findById(requestId).orElse(null);
+        if (vpSubmissionResult == null || vpSubmissionResult.getSubmissionStatus() == SubmissionStatus.FAILED){
+            return new VpTokenResultDto(transactionId,SubmissionStatus.FAILED,null);
+        }
+        JSONArray verifiableCredentials = new JSONObject(vpSubmissionResult.getVpToken()).getJSONArray(Constants.KEY_VERIFIABLE_CREDENTIAL);
+        List<VcResult> vcVerificationResults = getVcVerificationResults(verifiableCredentials);
+        return new VpTokenResultDto(transactionId,SubmissionStatus.SUCCESS,vcVerificationResults);
+    }
+
+    private static List<VcResult> getVcVerificationResults(JSONArray verifiableCredentials) {
+        List<VcResult> verificationResults = new ArrayList<>();
+        for (Object verifiableCredential : verifiableCredentials) {
+            JSONObject credential = new JSONObject((String) verifiableCredential).getJSONObject(Constants.KEY_VERIFIABLE_CREDENTIAL).getJSONObject(Constants.KEY_CREDENTIAL);
+            VerificationResult verificationResult = new CredentialsVerifier().verify(credential.toString(), CredentialFormat.LDP_VC);
```

Comment:
Lets understand the performance impacts of CredentialsVerifier instantiated everytime.

Reply:
Moved to singleton